pkgs/development/ocaml-modules/cudf/default.nix

```diff
@@ -0,0 +1,52 @@
+{ lib, fetchurl, stdenv, ocaml, ocamlbuild, findlib, ocaml_extlib, glib, perl, pkg-config, stdlib-shims, ounit }:
+
+stdenv.mkDerivation {
+  pname = "ocaml${ocaml.version}-cudf";
+  version = "0.9";
+
+  src = fetchurl {
+    url = "https://gforge.inria.fr/frs/download.php/36602/cudf-0.9.tar.gz";
+    sha256 = "sha256-mTLk2V3OI1sUNIYv84nM3reiirf0AuozG5ZzLCmn4Rw=";
+  };
+
+  buildFlags = [
+    "all"
+    "opt"
+  ];
+  buildInputs = [
+    findlib
+    glib
+    ocaml
+    ocaml_extlib
```

Comment:
My struggle to know what should be propagated or not is because I don't have a test application for all these dependencies I'm porting, once my target (Esy) builds, I can make all the optimizations and add tests and make a new PR handling it. I understand this is not ideal. But I think it is close. If I had not broken down the dependencies into different PRs, it would be a huge PR and I dread that.

Thanks for all the patience you are having with me. It is noted and appreciated.